modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/semantic_search/js/configuration/index.js

```diff
@@ -57,166 +93,221 @@ const transformToLabelValueArray = (items = {}) => {
  * This can be found on: System Settings > Search Experiences > Semantic Search
  */
 export default function ({
-	assetEntryClassNames,
 	availableAssetEntryClassNames,
 	availableEmbeddingVectorDimensions,
 	availableLanguageDisplayNames,
 	availableTextEmbeddingProviders,
 	availableTextTruncationStrategies,
-	cacheTimeout = '',
-	embeddingVectorDimensions,
-	huggingFaceAccessToken,
+	initialTextEmbeddingProviderConfigurations,
+	initialTextEmbeddingsEnabled,
 	learnMessages,
-	languageIds,
-	maxCharacterCount = '',
-	model,
-	modelTimeout = '',
 	namespace = '',
-	textEmbeddingProvider = TEXT_EMBEDDING_PROVIDER_TYPES.HUGGING_FACE_INFERENCE_API,
-	textEmbeddingsEnabled,
-	textTruncationStrategy,
-	txtaiHostAddress,
-	txtaiPassword,
-	txtaiUsername,
 }) {
-	const _handleFormikValidate = (values) => {
-		const errors = {};
-
-		// Validate "Asset Entry Class Names" field.
-
-		if (!values.assetEntryClassNames?.length) {
-			errors.assetEntryClassNames = sub(
-				Liferay.Language.get('the-x-field-is-required'),
-				[Liferay.Language.get('asset-entry-class-names')]
-			);
-		}
-
-		// Validate "Cache Timeout" field.
-
-		if (values.cacheTimeout === '') {
-			errors.cacheTimeout = Liferay.Language.get(
-				'this-field-is-required'
-			);
-		}
-		else {
-			if (values.cacheTimeout < 0) {
-				errors.cacheTimeout = sub(
-					Liferay.Language.get(
-						'please-enter-a-value-greater-than-or-equal-to-x'
-					),
-					['0']
-				);
-			}
-		}
-
-		// Validate "Hugging Face Access Token" field.
-
-		if (values.huggingFaceAccessToken === '') {
-			errors.huggingFaceAccessToken = Liferay.Language.get(
-				'this-field-is-required'
-			);
-		}
-
-		// Validate "Language IDs" field.
-
-		if (!values.languageIds?.length) {
-			errors.languageIds = sub(
-				Liferay.Language.get('the-x-field-is-required'),
-				[Liferay.Language.get('language-ids')]
-			);
-		}
-
-		// Validate "Max Character Count" field.
-
-		if (!values.maxCharacterCount === '') {
-			errors.maxCharacterCount = Liferay.Language.get(
-				'this-field-is-required'
-			);
-		}
-		else {
-			if (values.maxCharacterCount < 50) {
-				errors.maxCharacterCount = sub(
-					Liferay.Language.get(
-						'please-enter-a-value-greater-than-or-equal-to-x'
-					),
-					['50']
-				);
-			}
-
-			if (values.maxCharacterCount > 10000) {
-				errors.maxCharacterCount = sub(
-					Liferay.Language.get(
-						'please-enter-a-value-less-than-or-equal-to-x'
-					),
-					['10000']
-				);
-			}
-		}
-
-		// Validate "Model" field.
-
-		if (values.model === '') {
-			errors.model = Liferay.Language.get('this-field-is-required');
-		}
+	const initialTextEmbeddingProviderConfigurationsRef = useRef(
+		parseJSONString(initialTextEmbeddingProviderConfigurations)
+	);
```

Comment:
If `semanticSearchCompanyConfigurationDisplayContext.getTextEmbeddingProviderConfigurations()` can return an array, we can cover both array and string cases by updating this line to:

```suggestion
	const initialTextEmbeddingProviderConfigurationsRef = useRef(
		Array.isArray(initialTextEmbeddingProviderConfigurations) ? initialTextEmbeddingProviderConfigurations : parseJSONString(initialTextEmbeddingProviderConfigurations)
	);
```